warehouse/index.js

```diff
@@ -29,7 +31,17 @@ const timestampRegex = new RegExp(
 );
 
 function validTimestamp(input) {
-  return timestampRegex.test(input);
+  if (timestampRegex.test(input)) {
+    // check if date value lies in between min time and max time. if not then it's not a valid timestamp
+    const date = new Date(input).toISOString();
+    if (
+      Date.parse(minTimestamp) <= Date.parse(date) &&
```

Comment:
We don't need to parse minTimestamp and maxTimestamp for every timestamp field. Precompute them at top of the file. This gets called for every event's timestamp fields, let's try to reduce the computation.

Don't need to parse `Date.parse(date)` twice.